controllers/drplacementcontrol_controller.go

```diff
@@ -1264,9 +1264,53 @@ func (r *DRPlacementControlReconciler) finalizeDRPC(ctx context.Context, drpc *r
 		return fmt.Errorf("VRG adoption in progress")
 	}
 
-	// delete manifestworks (VRGs)
+	// delete manifestworks (VRGs and NSs)
 	for _, drClusterName := range rmnutil.DRPolicyClusterNames(drPolicy) {
-		err := mwu.DeleteManifestWorksForCluster(drClusterName)
+		// delete VRG manifestwork
+		mwName := mwu.BuildManifestWorkName(rmnutil.MWTypeVRG)
+		mwNamespace := drClusterName
+
+		err = mwu.DeleteManifestWork(mwName, mwNamespace)
+		if err != nil {
+			return fmt.Errorf("%w", err)
+		}
+
+		// check if the Spec has deleteOption. If not call createOrUpdate() to get
+		// the deleteOption added.
+
+		// delete NS manifestwork
+		mwName = mwu.BuildManifestWorkName(rmnutil.MWTypeNS)
+		mwNamespace = drClusterName
+
+		mw := &ocmworkv1.ManifestWork{}
+		err := mwu.Client.Get(mwu.Ctx, types.NamespacedName{Name: mwName, Namespace: mwNamespace}, mw)
+		
+		if err != nil {
+			if errors.IsNotFound(err) {
+				return nil
```

Comment:
Cannot return from here, as that would signal that finalize DRPC is a success. It should continue with other cleanup below in this case.